src/map/utils/battleutils.cpp

```diff
@@ -730,6 +730,16 @@ namespace battleutils
 
                         if (PDefender->isAlive())
                         {
+                            // Subpower is the remaining damage that can be drained. When it reaches 0 the effect ends
+                            int remainingDrain = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)->GetSubPower();

```

Comment:
I'm not sure what you're getting at with spikes being set by other things because wouldn't a mob having auto-spikes (if there's one with auto Dread Spikes, pretty sure there's no mob on retail with Reprisal in any form...) not have a limited amount of draining capacity?

So I should change something with this?